File: backend/internal/constants/report.go
package constants

// ReportReason defines product report reason enum values shared with the frontend.
const (
	ReportReasonFalseDescription = "false_description"
	ReportReasonProhibitedItem   = "prohibited_item"
	ReportReasonFraud            = "fraud"
	ReportReasonOther            = "other"
)

// ReportReasons lists all valid product report reasons.
var ReportReasons = []string{
	ReportReasonFalseDescription, ReportReasonProhibitedItem, ReportReasonFraud, ReportReasonOther,
}

// IsReportReason reports whether the given reason is valid.
func IsReportReason(r string) bool {
	for _, v := range ReportReasons {
		if v == r {
			return true
		}
	}
	return false
}

// ReportReasonText returns the Chinese label of a report reason.
func ReportReasonText(r string) string {
	switch r {
	case ReportReasonFalseDescription:
		return "虚假描述"
	case ReportReasonProhibitedItem:
		return "违禁物品"
	case ReportReasonFraud:
		return "疑似诈骗"
	case ReportReasonOther:
		return "其他"
	default:
		return "未知"
	}
}

// ReportStatus defines report workflow states shared with the frontend.
const (
	ReportStatusPending   = "pending"
	ReportStatusTakenDown = "taken_down"
	ReportStatusRejected  = "rejected"
)

// ReportStatuses lists all valid report statuses.
var ReportStatuses = []string{
	ReportStatusPending, ReportStatusTakenDown, ReportStatusRejected,
}

// IsReportStatus reports whether the given status is valid.
func IsReportStatus(s string) bool {
	for _, v := range ReportStatuses {
		if v == s {
			return true
		}
	}
	return false
}

// ReportStatusText returns the Chinese label of a report status.
func ReportStatusText(s string) string {
	switch s {
	case ReportStatusPending:
		return "待处理"
	case ReportStatusTakenDown:
		return "已下架"
	case ReportStatusRejected:
		return "已驳回"
	default:
		return "未知"
	}
}

// ReportAction defines admin handling actions shared with the frontend.
const (
	ReportActionTakeDown = "take_down"
	ReportActionReject   = "reject"
)

// ReportActions lists all valid admin report actions.
var ReportActions = []string{ReportActionTakeDown, ReportActionReject}

// IsReportAction reports whether the given action is valid.
func IsReportAction(a string) bool {
	for _, v := range ReportActions {
		if v == a {
			return true
		}
	}
	return false
}

// ReportActionText returns the Chinese label of an admin report action.
func ReportActionText(a string) string {
	switch a {
	case ReportActionTakeDown:
		return "下架商品"
	case ReportActionReject:
		return "驳回举报"
	default:
		return "未知"
	}
}

File: frontend/src/constants/report.ts
export const REPORT_REASONS = [
  { value: 'fake_description', label: '虚假描述' },
  { value: 'prohibited_item', label: '违禁物品' },
] as const

export const REPORT_STATUSES = [
  { value: 'pending', label: '待处理', type: 'warning' },
  { value: 'taken_down', label: '已下架', type: 'danger' },
  { value: 'rejected', label: '已驳回', type: 'info' },
] as const

export const REPORT_ACTIONS = [
  { value: 'take_down', label: '下架商品', type: 'danger' },
  { value: 'reject', label: '驳回举报', type: 'info' },
] as const

export function reportReasonLabel(value: string): string {
  return REPORT_REASONS.find((r) => r.value === value)?.label ?? value
}

export function reportStatusLabel(value: string): string {
  return REPORT_STATUSES.find((s) => s.value === value)?.label ?? value
}

export function reportStatusType(value: string): string {
  return REPORT_STATUSES.find((s) => s.value === value)?.type ?? 'info'
}
